src/node_main_instance.cc

```diff
@@ -38,12 +41,20 @@ NodeMainInstance::NodeMainInstance(const SnapshotData* snapshot_data,
     : args_(args),
       exec_args_(exec_args),
       array_buffer_allocator_(ArrayBufferAllocator::Create()),
+      cpp_heap_(CppHeap::Create(
+          platform,
+          CppHeapCreateParams{
+              {},
+              WrapperDescriptor(BaseObject::kEmbedderType,
+                                BaseObject::kSlot,
+                                BaseObject::kDefaultCppGCEmebdderTypeID)})),
```

Comment:
How are embedders expected to get the correct values such as `kDefaultCppGCEmebdderTypeID`? Shouldn't this be part of `NewIsolate()` or `SetIsolateCreateParamsForNode()`?

Reply:
Actually, wrapper descriptors are going to be deprecated so that embedders don't have to care about what the value is: https://github.com/nodejs/node-v8/issues/283.

Read more at https://docs.google.com/document/d/1-sBltmlx4yUIzmNHrRO9lKjqiN-uJ81gCmj9-Gq5d5w/edit

Reply:
Okay, then this will 'just' attach a CppHeap with default options in the future anyway? That still _feels_ like `NewIsolate()` or `SetIsolateCreateParamsForNode()` would be the right places for this, as the goal is to initialize a 'Node.js-flavored' `Isolate` instance, unless I'm missing something

Reply:
Yes, I turned this into a draft because I think the current form is not complete.